### ISSUE TITLE
offline operation impossible with pypi dependencies

### ISSUE THREAD

**Issue by akhmerov** [prefix-dev/pixi#2445]
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
mkdir pixi_bug && cd pixi_bug
pixi init
pixi add python
pixi add --pypi requests
# TURN OFF INTERNET CONNECTION
pixi clean cache
pixi run --no-lockfile-update --frozen python # Fails to connect to pypi
```

### Issue description

It seems that it is impossible to use a pixi project without internet connection if this project has pypi dependencies. Even with `--no-lockfile-update --frozen` pixi will sometimes try to connect to pypi and fail.

### Expected behavior

That it is possible to run code without internet connection 🙃

**Comment by petrushy** [comment]
+1 had same problem on a flight last night.

**Comment by ruben-arts** [comment]
Thanks for the report! We'll have to look into this. PR's are always welcome!

**Comment by akhmerov** [comment]
@ruben-arts, @tdejager, I still see the issue in pixi 0.36, please reopen.

**Comment by tdejager** [comment]
Could you paste the exact error you are getting, after the last step?

**Comment by ruben-arts** [comment]
Tried the example again and it is still broken on `v0.35.0` but fixed on `v0.36.0`.

**Comment by akhmerov** [comment]
I'm sorry. It also works for me now; not sure what was wrong yesterday.
